wrappers/src/object_cs.cpp

```diff
@@ -280,11 +281,21 @@ extern "C" {
         });
     }
 
-    REALM_EXPORT ManagedNotificationTokenContext* object_add_notification_callback(Object* object, void* managed_object, NativeException::Marshallable& ex)
+    REALM_EXPORT ManagedNotificationTokenContext* object_add_notification_callback(Object* object, void* managed_object, size_t* property_indices, size_t property_count, NativeException::Marshallable& ex)
     {
+        TableKey tk = object->get_object_schema().table_key;
+        ColKey ck;
+        KeyPathArray keyPathArray;
+        for (int i = 0; i < property_count; i++) {
+            KeyPath keyPath;
+            auto prop = get_property(*object, property_indices[i]);
+            ck = prop.column_key;
+            keyPath.push_back(std::make_pair(tk, ck));
+            keyPathArray.push_back(keyPath);
```

Comment:
Do you need to create `keyPath` at all? Can't you just pass `std::make_pair` to the `push_back` method? Not very good with C++. so it's more of a curiosity 😁

Reply:
If I'm not mistaken we cannot bypass this, `KeyPath` and `KeyPathArray` are defined ass 

`using KeyPath = std::vector<std::pair<TableKey, ColKey>>;`
`using KeyPathArray = std::vector<KeyPath>;`

`add_notification_callback` takes a nested vector since you can provide multiple keyPaths at once if you do it manually (example: "foo.bar.prop1", "foo.bar.prop2". In our case the `keyPath` doesn't really make sense since we only add one element to it (that would be "foo" with the same example). But it's mostly there now to comply with the parameter we need to provide. In the future it will also make more sense once we pass keyPaths ourself